src/utils/ucc_byte_order.h

```diff
@@ -0,0 +1,21 @@
+/**
+ * Copyright (C) Mellanox Technologies Ltd. 2020-2021.  ALL RIGHTS RESERVED.
+ * See file LICENSE for terms.
+ */
+
+//TODO was taken from here, rights issue?
+// https://github.com/tensorflow/tensorflow/blob/master/tensorflow/core/platform/byte_order.h
+
+#ifndef UCC_BYTE_ORDER_H_
+#define UCC_BYTE_ORDER_H_
+
+// Byte order defines provided by gcc. MSVC doesn't define those so
+// we define them here.
+// We assume that all windows platform out there are little endian.
+#if defined(_MSC_VER) && !defined(__clang__)
```

Comment:
What this really do? adds some defins for MS Visual Studio compiler? @manjugv are we even supporting MS ?
Regardless, we don't need that header at all. Instead what we need to do is to detect ENDIANESS during configure step. There is a autotool way to do that:
https://www.gnu.org/software/autoconf/manual/autoconf-2.65/html_node/C-Compiler.html

AC_C_BIGENDIAN

Autotool will run the check during configure, we will define macro in config.h (UCC_BIG_ENDIAN vs UCC_LITTLE_ENDIAN) and use those macros in the code.